src/components/advanced-audio-desc/advanced-audio-desc.tsx

```diff
@@ -29,12 +32,38 @@ const mapStateToProps = state => ({
  * @extends {Component}
  */
 @connect(mapStateToProps, bindActions(actions))
+@withPlayer
 @withEventDispatcher(COMPONENT_NAME)
 @withText({AdvancedAudioDescText: 'settings.advancedAudioDescription'})
-class AdvancedAudioDesc extends Component<any, any> {
+class AdvancedAudioDesc extends Component<any, any> implements IconComponent {
   constructor(props: any) {
     super();
     this.state = {toggle: false};
+    registerToBottomBar(COMPONENT_NAME, props.player, () => this.registerComponent());
+  }
+
+  registerComponent(): any {
+    return {
+      ariaLabel: () => this.getComponentText(),
+      displayName: COMPONENT_NAME,
+      order: 5,
+      svgIcon: () => this.getSvgIcon(),
+      onClick: () => this.onClick(),
+      component: () => {
+        return getComp({...this.props, classNames: [style.upperBarIcon]});
+      },
+      selfManagement: true
+    };
+  }
+
+  getComponentText = (): any => {
+    return this.props.AdvancedAudioDescText;
+  }
+
+  getSvgIcon = (): any => {
+    return {
+      type: redux.useStore().getState().settings.advancedAudioDesc ? IconType.AdvancedAudioDescriptionActive : IconType.AdvancedAudioDescription
```

Comment:
there's probably a better way to fetch value from state...
check what the best practice is (for example, theres an option to have a file full of getters that just return stuff from state). also, maybe getMemo can be used here.